### ISSUE TITLE
Reported issue with rustup on Ubuntu 20.04, disabling configurations

### ISSUE THREAD

**Issue by skade** [rust-lang/rustup#2646]
**Problem**
We got a report over at https://github.com/rust-lang/www.rust-lang.org/issues/1423 for what seems to be a rustup behaviour where rustup configuration disables other configs on Ubuntu 20.04

I'm a little hesitant to copy it over here, as I'm not quite sure which parts of the text are relevant.
I think it's more effective if someone more acquainted with rustup/ubuntu took lead here.

**Comment by douglaslwatts** [comment]
Here are the relevant parts of the initial bug report I made:

I installed Rust using the script acquired from the curl command recommended on your website: https://www.rust-lang.org/tools/install . As soon as I logged out and back in, ALL of my custom settings, keyboard shortcuts were completely gone and gnome-shell-extensions turned off ...literally every single setting I have ever changed on my system. I reset them all within about 30 minutes, not too big a deal but unusual and quite annoying. When I rebooted, same thing ...every time I rebooted all settings and everything I customize was gone. Also, my custom mapping of caps-lock as ctrl and both shift keys together enable caps-lock was gone and I had to reset that. While resetting all of the settings a second time, which only took about 10 minutes since I remembered where everything was due to just having done so, I noticed that as I set keyboard shortcuts some would be removed immediately and others duplicated. Sorry, but I already restored my /home/me from the snapshot that was taken this morning so I cannot try to find what files were overwritten. I reported the issue in the wrong place because the website: https://www.rust-lang.org/tools/install links issues there at the bottom of the page when your click "File an issue". I chose the default install option when running the install script.

OS:                           Ubuntu 20.04.2 LTS
OS type:                   64-bit
Processor:                Intel Core i7-7700K CPU 4.20GHz x 8
Gnome Version:       3.36.8
Windowing System:  X11
Graphics:                  Mesa Intel HD Graphics 630 (KBL GT2)

**Comment by rbtcollins** [comment]
I'm afraid it is going to be very hard to diagnose this. rustup does write to files in `~` naturally, but without seeing the faulting files we can only speculated and go and look hard at our code  / run fuzz experiments etc. And we sortof have a natural laboratory in all the users such as yourself using it - but this is the first report of this kind, and until we have a second we're probably going to be stumped.  @kinnison your thoughts?

**Comment by douglaslwatts** [comment]
considering that I keep snapshots and it is easy to restore from them, if I were willing to attempt to reproduce the issue on my laptop, which is set up almost identically to my desktop and has "similar" hardware, what would I need to do provide you to assist in tracking down the problem? I don't use my laptop very much as I much prefer my desktop.

**Comment by douglaslwatts** [comment]
I could also just try to reproduce the issue on the same machine, my desktop, since I can restore in less than a minute with rsync anyway.

**Comment by douglaslwatts** [comment]
I planned to reproduce the issue, rsync ~ to somewhere, use rsync to restore ~ from my snapshot, but have it be verbose and redirect the output to a file, and then provide you with that file along with the files that were changed in the restore. Unfortunately, the issue was not reproduced on my laptop and the installation script worked perfectly. I tried reproducing it on my desktop, the original machine, and same results. The installer worked as expected and no issue. On the up side I have latest Rust using the installation method recommended on https://www.rust-lang.org/tools/install and VSCode is successfully using the language server on both machines. Wish I had thought to do the above procedure before restoring this morning, but I just needed to get some school work done. If I have any issues in the future when I update with "rustup update", I will do the above procedure. Thanks so much for the extremely quick response : )
